Error while using `smalrye.jwt.sign.key.location` properties with file: scheme

I have an application using quarkus-jwt with the following configurations:

```properties
mp.jwt.verify.publickey.location=file:/keys/privateKey.pem
```

The file is on `/keys/' directory.

With this configuration I am receiving the following error:

```bash
2023-11-12 16:22:21,400 INFO  [io.quarkus] (main) condomind-api 1.0.0-SNAPSHOT on JVM (powered by Quarkus 3.5.0) started in 2.953s. Listening on: http://0.0.0.0:8080
2023-11-12 16:22:21,401 INFO  [io.quarkus] (main) Profile prod activated. 
2023-11-12 16:22:21,401 INFO  [io.quarkus] (main) Installed features: [agroal, amazon-ses, cdi, hibernate-orm, hibernate-orm-panache, hibernate-validator, jdbc-mysql, narayana-jta, qute, resteasy-reactive, resteasy-reactive-jackson, resteasy-reactive-qute, security, smallrye-context-propagation, smallrye-jwt, smallrye-openapi, vertx]
2023-11-12 16:22:28,844 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /auth/sign-in failed, error id: caf298a0-ac90-4ec7-aabb-5400a63c14f9-1: io.smallrye.jwt.build.JwtSignatureException: SRJWT05009: 
        at io.smallrye.jwt.build.impl.JwtSignatureImpl.sign(JwtSignatureImpl.java:105)
        at com.github.mcruzdev.infrastructure.security.JwtAdapter.generate(JwtAdapter.java:17)
        at com.github.mcruzdev.application.usecases.auth.Login.execute(Login.java:28)
        at com.github.mcruzdev.application.usecases.auth.Login_ClientProxy.execute(Unknown Source)
        at com.github.mcruzdev.presentation.resources.AuthResource.signIn(AuthResource.java:60)
        at com.github.mcruzdev.presentation.resources.AuthResource$quarkusrestinvoker$signIn_a406def941aac544eb675e8bd55eaf7594b3a996.invoke(Unknown Source)
        at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.InputStream.read(byte[])" because "is" is null
        at io.smallrye.jwt.util.ResourceUtils.readBytes(ResourceUtils.java:93)
        at io.smallrye.jwt.build.impl.JwtSignatureImpl.sign(JwtSignatureImpl.java:82)
        ... 15 more
``` 

I do not know but the error can starts from [here](https://github.com/smallrye/smallrye-jwt/blob/66a74d5e0b4cdf58977d9d3968d0354404625d13/implementation/common/src/main/java/io/smallrye/jwt/util/ResourceUtils.java#L73)

Hi @sberyozkin TY for the comment.

I changed to `/keys/privateKey.pem` and I am receiving it:

```bash
Caused by: java.net.MalformedURLException: no protocol: /keys/privateKey.pem
```

--- a/implementation/common/src/main/java/io/smallrye/jwt/util/JWTUtilLogging.java
+++ b/implementation/common/src/main/java/io/smallrye/jwt/util/JWTUtilLogging.java
@@ -38,4 +38,8 @@ interface JWTUtilLogging extends BasicLogger {
     @LogMessage(level = Logger.Level.DEBUG)
     @Message(id = 1009, value = "Failed to parse the JWK JSON representation")
     void parsingJwksFailed();
+
+    @LogMessage(level = Logger.Level.DEBUG)
+    @Message(id = 1010, value = "File %s is not found")
+    void fileIsNotFound(String fileLocation);
 }

--- a/implementation/common/src/main/java/io/smallrye/jwt/util/KeyUtils.java
+++ b/implementation/common/src/main/java/io/smallrye/jwt/util/KeyUtils.java
@@ -296,8 +296,11 @@ public final class KeyUtils {
                 ? KeyStore.getInstance(theKeyStoreType, provider)
                 : KeyStore.getInstance(theKeyStoreType);
         if (keyStorePath != null) {
-            try (InputStream is = ResourceUtils.getResourceStream(keyStorePath)) {
-                keyStore.load(is, keyStorePassword.toCharArray());
+            InputStream is = ResourceUtils.getResourceStream(keyStorePath.trim());
+            if (is != null) {
+                try (InputStream keyStream = is) {
+                    keyStore.load(keyStream, keyStorePassword.toCharArray());
+                }
             }
         } else {
             keyStore.load(null, keyStorePassword.toCharArray());

--- a/implementation/common/src/main/java/io/smallrye/jwt/util/ResourceUtils.java
+++ b/implementation/common/src/main/java/io/smallrye/jwt/util/ResourceUtils.java
@@ -104,6 +104,7 @@ public final class ResourceUtils {
         try {
             return new FileInputStream(publicKeyLocation);
         } catch (FileNotFoundException e) {
+            JWTUtilLogging.log.fileIsNotFound(publicKeyLocation);
             return null;
         }
     }

--- a/implementation/jwt-build/src/main/java/io/smallrye/jwt/build/impl/JwtEncryptionImpl.java
+++ b/implementation/jwt-build/src/main/java/io/smallrye/jwt/build/impl/JwtEncryptionImpl.java
@@ -81,8 +81,11 @@ class JwtEncryptionImpl implements JwtEncryptionBuilder {
             if (keyLocation != null) {
                 key = JwtBuildUtils.readPublicKeyFromKeystore(keyLocation.trim());
                 if (key == null) {
-                    try (InputStream keyStream = ResourceUtils.getResourceStream(keyLocation.trim())) {
-                        key = getEncryptionKeyFromKeyContent(new String(ResourceUtils.readBytes(keyStream)));
+                    InputStream is = ResourceUtils.getResourceStream(keyLocation.trim());
+                    if (is != null) {
+                        try (InputStream keyStream = is) {
+                            key = getEncryptionKeyFromKeyContent(new String(ResourceUtils.readBytes(keyStream)));
+                        }
                     }
                 }
             } else {

--- a/implementation/jwt-build/src/main/java/io/smallrye/jwt/build/impl/JwtSignatureImpl.java
+++ b/implementation/jwt-build/src/main/java/io/smallrye/jwt/build/impl/JwtSignatureImpl.java
@@ -78,8 +78,11 @@ class JwtSignatureImpl implements JwtSignature {
                 if (keyLocation != null) {
                     key = JwtBuildUtils.readPrivateKeyFromKeystore(keyLocation.trim());
                     if (key == null) {
-                        try (InputStream keyStream = ResourceUtils.getResourceStream(keyLocation.trim())) {
-                            key = getSigningKeyFromKeyContent(new String(ResourceUtils.readBytes(keyStream)));
+                        InputStream is = ResourceUtils.getResourceStream(keyLocation.trim());
+                        if (is != null) {
+                            try (InputStream keyStream = is) {
+                                key = getSigningKeyFromKeyContent(new String(ResourceUtils.readBytes(keyStream)));
+                            }
                         }
                     }
                 } else {

--- a/implementation/jwt-build/src/test/java/io/smallrye/jwt/build/JwtSignPS256Test.java
+++ b/implementation/jwt-build/src/test/java/io/smallrye/jwt/build/JwtSignPS256Test.java
@@ -18,6 +18,7 @@ package io.smallrye.jwt.build;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
+import java.io.File;
 import java.security.Security;
 
 import org.jose4j.jws.JsonWebSignature;
@@ -56,10 +57,12 @@ class JwtSignPS256Test {
 
     @Test
     void signClaimsPS256() throws Exception {
+        String path = "src/test/resources/privateKey.pem";
+        File file = new File(path);
         String jwt = Jwt.claims()
                 .claim("customClaim", "custom-value")
                 .jws().algorithm(SignatureAlgorithm.PS256)
-                .sign("/privateKey.pem");
+                .sign("file:" + file.getAbsolutePath());
 
         JsonWebSignature jws = JwtSignTest.getVerifiedJws(jwt, KeyUtils.readPublicKey("/publicKey.pem"));
         JwtClaims claims = JwtClaims.parse(jws.getPayload());

